go-controller/pkg/ovn/controller/apbroute/external_controller_pod.go

```diff
@@ -28,6 +30,13 @@ func (m *externalPolicyManager) syncPod(pod *corev1.Pod, routeQueue workqueue.Ty
 }
 
 func getExGwPodIPs(gatewayPod *corev1.Pod, networkName string) (sets.Set[string], error) {
+	// If an external gateway pod is in terminating or not ready state then don't return the
+	// IPs for the external gateway pod
+	if util.PodTerminating(gatewayPod) || util.PodNotReady(gatewayPod) {
```

Comment:
isn't pod not running in both cases ? if that is the case, can't just use `!util.PodRunning(gatewayPod)` ?

Reply:
In both the cases, pod can remain in running state thus using `!util.PodRunning(gatewayPod)` may not have the desired result.

When the pod is marked for deletion with a finalizer, the pod will continue to remain in Running state up until the finalizer is removed from the pod. The deletion timestamp will be added to the pod in a such case.

When the readiness probe fails for some reason, the pod will be marked as not ready. However, the pod can remain in Running state even after being marked as not ready.

Reply:
Even if the gateway pod is in running state while it's terminating or in not ready state, it may not be able to serve traffic. In terminating state, it might be gracefully shutting down the processes running inside the pod and during that time it may not serve the traffic. In the case of gateway not being ready, there are chances that even if the pod is still in running state it may not serve the traffic as the readiness probe has failed. Thus, rather than waiting for the pod to completely stop running, it's being preemptively removed from the list of active gateways to avoid the scenarios where a gateway is marked as active however it does not serve the traffic that is being forwarded to it.